whylog/assistant/regex_assistant/regex_match.py

```diff
@@ -44,11 +45,18 @@ def update_by_regex(self, new_regex):
         Updates self.param_groups so that they correspond to new_regex groups
         """
 
-        groups = regex_groups(new_regex, self.line_text)
-
         if new_regex[-1] != '$':
             new_regex += '$'
 
+        try:
+            verify_regex(new_regex, self.line_text)
+        except NotMatchingRegexError:
+            self.regex = new_regex
+            self.param_groups = {}
+            self.primary_key = []
+            return
```

Comment:
In current version, `self.regex = new_regex` is copied and it makes this code brittle and hard to understand. While syntax suggested by Andrzej is rarely used, it was put in the python language so that we can utilize it in a case like this rather than copying code.